stdlib/list.mli

```diff
@@ -73,6 +73,14 @@ val nth_opt: 'a list -> int -> 'a option
 val rev : 'a list -> 'a list
 (** List reversal. *)
 
+val init : int -> (int -> 'a) -> 'a list
+(** [List.init n f] returns a fresh list of length [n],
+   with element number [i] initialized to the result of [f i].
+   In other terms, [List.init n f] tabulates the results of [f]
+   applied to the integers [0] to [n-1].
+
+   Raise [Invalid_argument] if [n < 0].*)
```

Comment:
Sorry for more comments on this comment. I think the copy-paste from Array has lost sight of how this function should be documented, especially now that it builds the list in reverse.

```ocaml
(** [List.init n f] is [f 0; f 1; ... ; f (n - 1)].

    @raise Invalid_argument if [n < 0].

    @since 4.05.0
*)
```

I can't decide whether we should be explicit that the list is constructed in reverse - I *think* that the order of evaluation of `[f 0; f 1]` is itself not specified in OCaml, which would imply that not mentioning it is OK?